sdk/storage/Azure.Storage.Files/src/FileClient.cs

```diff
@@ -2527,7 +2597,8 @@ private void SetNameFieldsIfNull()
                         marker: marker,
```

Comment:
This needs to change.  We're going to wrap this in a loop that continues executing until no additional `marker` is returned or it throws.